lib/rules/unit-case/index.js

```diff
@@ -57,6 +58,23 @@ const rule = function(expectation) {
           result,
           ruleName
         });
+
+        if (context.fix) {
+          const source = node.name === "media" ? node.params : node.value;
+
+          styleSearch({ source, target: unit }, match => {
+            const stringStart = source.slice(0, match.startIndex);
+            const stringEnd = source.slice(match.endIndex, source.length);
+
+            const res = `${stringStart}${expectedUnit}${stringEnd}`;
+
+            if (node.name === "media") {
+              node.params = res;
+            } else {
+              node.value = res;
+            }
+          });
+        }
```

Comment:
Move it before `report` and add `return` at the end of `if` statement. This way we won't show rule violation when we fixed the problem.